### DISCUSSION TITLE
How to login and logout?

### DISCUSSION THREAD

**Discussion by summerisgood1** [gladly-team/next-firebase-auth#185] [Q&A]
I saw the example folder using firebaseui. How about I use custom login component? How to login and logout?

**Comment by shreyas-jadhav** [comment]
This would happen automatically when the auth state changes, assuming you are actually logging in and signing out via Firebase SDK methods.

**Comment by summerisgood1** [comment]
So I need to use SDK methods too, then What is the function on api/login and api/logout?

**Comment by shreyas-jadhav** [comment]
That's for server side, setting and unsetting the cookies. They are called automatically, when users signs in or signs out, if I'm not wrong.

**Comment by Mrjuanblack** [comment]
Anyone got example code on how to achieve this?

**Reply by Mrjuanblack** [reply]
I found a simple example on this issue on how to handle login with a custom component: https://github.com/gladly-team/next-firebase-auth/issues/591#issue-1484731729
